Fixed the Multifarm-Support for the Undead Sapling

RedEtherbloom opened the pull request:
(See [here](https://github.com/CyclopsMC/EvilCraft-Compat/issues/3))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/CyclopsMC/EvilCraft-Compat?pullRequest=2) <br/>All committers have signed the CLA.

josephcsible commented:
Why did you close this?

RedEtherbloom commented:
I wanted to implement a cleaner way of this, because this way was rather ugly.
I will upload a better version of the code later. So i closed it that it won't be considered for merging. 

Sorry for wasting your time.

rubensworks commented on src/main/java/org/cyclops/evilcraftcompat/modcompat/forestry/FarmableUndeadSapling.java:
```diff
@@ -0,0 +1,87 @@
+import org.cyclops.evilcraft.block.UndeadSaplingConfig;
+import org.cyclops.evilcraft.item.HardenedBloodShardConfig;
+
+public class FarmableUndeadSapling implements IFarmable {
```
Why is this class needed? Can't you just use Forestry's `FarmableSapling` class directly?

RedEtherbloom replied:
No, I can't. You're build script uses an old version of forestry, which does not implement FarmableSapling yet.

rubensworks replied:
Perhaps it's a good time to update then, as the `add-farmable-sapling` message is broken anyways.

rubensworks commented on src/main/java/org/cyclops/evilcraftcompat/modcompat/forestry/ForestryModCompat.java:
```diff
@@ -28,10 +29,10 @@ public String getModID() {
-	            FMLInterModComms.sendMessage(getModID(), "add-farmable-sapling",
-						"farmArboreal@" + Block.REGISTRY.getNameForObject(UndeadSaplingConfig._instance.getBlockInstance()).toString() + ".0");
+	        if(Configs.isEnabled(UndeadSaplingConfig.class) && Configs.isEnabled(HardenedBloodShardConfig.class)) {
+                ForestryAPI.farmRegistry.registerFarmables("farmArboreal", new FarmableUndeadSapling());
```
This will unfortunately cause a crash when Forestry is not installed.

You'll have to move this `ForestryAPI` call to a separate class. Have a look at how this is done with other mod compats, such as Immersive Engineering.

Commit:
Updated the forestry version to the latest release and readded the undead sapling to the tree farm

RedEtherbloom commented:
I updated Forestry to the newest release, reimplemented my code in a cleaner way and cleaned up my commit mess.

Should i update the rest of the dependencies in Evilcraft (and Compat, and CylopsCore etc.) to the latest stable releases while I'm on it?

rubensworks commented:
Great, thanks!

> Should i update the rest of the dependencies in Evilcraft (and Compat, and CylopsCore etc.) to the latest stable releases while I'm on it?

No, no need for that.
